File: DSA11002.cpp
#include <iostream>
#include <deque>
#include <vector>
#include <algorithm>
#include <cstring>
#include <string>
#include <sstream>
using namespace std;
void function() {
    int n; cin >> n;
    deque<int> stack;
    string str;
    getline(cin >> ws, str);
    string i;
    stringstream ss(str);
    vector <string> formulation;
    while (ss >> i){
        formulation.push_back(i);
    }
    reverse(formulation.begin(), formulation.end());
    string operators = "+-*/";
    for (auto i: formulation){
        if (operators.find(i) == string::npos) {
            stack.push_back(stoi(i));
        } else {
            int operand2 = stack.front();
            stack.pop_front();
            int operand1 = stack.front();
            stack.pop_front();
            int result;
            if (i == "+") {
                result = operand1 + operand2;
            } else if (i == "-") {
                result = operand1 - operand2;
            } else if (i == "*") {
                result = operand1 * operand2;
            } else if (i == "/") {
                result = operand1 / operand2;
            }
            stack.push_back(result);
        }
    }
    cout << stack.front() << endl;
}

int main() {
    int t;
    cin >> t;
    for (int i = 0; i < t; ++i) {
        function();
    }
    return 0;
}

File: DSA11002.py
from collections import deque


def function(*line, **kwarg):
    n = int(input())
    formulation = input().split()[::-1]
    operator = ['+', '-', '*', '/']
    stack = deque()
    for i in formulation:
        if i not in operator:
            stack.append(i)
        else:
            operand2 = stack.popleft()
            operand1 = stack.popleft()
            if i == '/': 
                i = '//'
            stack.append(str(eval(operand1 + i + operand2)))
    print(stack[0])


if __name__ == "__main__":
    for i in range(int(input())):
        function()
